pkg/parser/gomod_test.go

```diff
@@ -0,0 +1,26 @@
+package parser
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var goDeps = []string{
+	"connectrpc.com/connect",  // Direct
+	"github.com/anchore/syft", // Direct
+	"github.com/gocql/gocql",  // Direct
+	"gopkg.in/inf.v0",         // indirect
+}
+
+func Test_GomodParser_Simple(t *testing.T) {
+	manifest, err := parseGoModFile("./fixtures/go/go.mod", &ParserConfig{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, 4, len(manifest.Packages))
```

Comment:
We need to retain the dependency relationship. In this example: 3 are direct and only 1 indirect.